src/components/agent-inbox/index.tsx

```diff
@@ -4,16 +4,82 @@ import { INBOX_PARAM, VIEW_STATE_THREAD_QUERY_PARAM } from "./constants";
 import { ThreadStatusWithAll } from "./types";
 import { AgentInboxView } from "./inbox-view";
 import { ThreadView } from "./thread-view";
+import { useScrollPosition } from "./hooks/use-scroll-position";
 
 export function AgentInbox<
   ThreadValues extends Record<string, any> = Record<string, any>,
 >() {
   const { searchParams, updateQueryParams, getSearchParam } = useQueryParams();
   const [selectedInbox, setSelectedInbox] =
     React.useState<ThreadStatusWithAll>("interrupted");
+  const { saveScrollPosition, restoreScrollPosition } = useScrollPosition();
+  const containerRef = React.useRef<HTMLDivElement>(null);
 
   const selectedThreadIdParam = searchParams.get(VIEW_STATE_THREAD_QUERY_PARAM);
   const isStateViewOpen = !!selectedThreadIdParam;
+  const prevIsStateViewOpen = React.useRef<boolean>(false);
```

Comment:
nit, it can infer from the default value
```suggestion
  const prevIsStateViewOpen = React.useRef(false);
```